caas/kubernetes/provider/provider.go

```diff
@@ -131,6 +131,21 @@ func CloudSpecToK8sRestConfig(cloudSpec environscloudspec.CloudSpec) (*rest.Conf
 	}, nil
 }
 
+// PatchCloudCredentialForCloudSpec patches the credential with the provided k8s secret token.
+func PatchCloudCredentialForCloudSpec(cred cloud.Credential, token string) (cloud.Credential, error) {
```

Comment:
See other comment - I think this should be a method on the provider itself, taking a service account as an arg

Reply:
As discussed, this function doesn't any dependency on the k8sclient, so it's better to keep it as a function.
Also, but it can be moved to `github.com/juju/juju/caas/kubernetes/cloud` package.